source/adapters/native_cpu/threadpool.hpp

```diff
@@ -208,7 +208,80 @@ template <typename ThreadPoolT> class threadpool_interface {
     return workerTask->get_future();
   }
 };
+using simple_threadpool_t = threadpool_interface<detail::simple_thread_pool>;
 
-using threadpool_t = threadpool_interface<detail::simple_thread_pool>;
+class TasksInfo_TP {
+  using FType = std::future<void>;
+  std::vector<FType> futures;
+public:
+  inline void schedule(FType &&f) { futures.emplace_back(std::move(f)); }
+  inline void wait() {
```

Comment:
Based on other APIs (e.g. Boost), this might possibly be clearer to rename to something like `wait_all` to highlight the difference between it and `future::wait`?